test/dsl/dsl_options.jl

```diff
@@ -424,6 +424,74 @@ let
     end
 end
 
+# Tests that explicitly declaring a single symbol as several things does not work.
+# Several of these are broken, but note sure how to test broken-ness on `@test_throws false Exception @eval`.
+# Relevant issue: https://github.com/SciML/Catalyst.jl/issues/1173
+let
+    # Species + parameter.
+    @test_broken false #@test_throws Exception @eval @reaction_network begin
+        #@species X(t)
+        #@parameters X
+    #end
+
+    # Species + variable.
+    @test_broken false #@test_throws Exception @eval @reaction_network begin
+        #@species X(t)
+        #@variables X(t)
+    #end
+
+    # Variable + parameter.
+    @test_broken false #@test_throws Exception @eval @reaction_network begin
+        #@variables X(t)
+        #@parameters X
+    #end
+
+    # Species + differential.
+    @test_throws Exception @eval @reaction_network begin
+        @species X(t)
+        @differentials X = Differential(t)
+    end
+
+    # Parameter + differential.
+    @test_throws Exception @eval @reaction_network begin
+        @parameters X
+        @differentials X = Differential(t)
+    end
+
+    # Variable + differential.
+    @test_throws Exception @eval @reaction_network begin
+        @variables X(t)
+        @differentials X = Differential(t)
+    end
+
+    # Parameter + observable (species/variable + observable is OK, as this e.g. provide additional observables information).
+    @test_broken false #@test_throws Exception @eval @reaction_network begin
```

Comment:
I am not really sure how to make a `@test_throws` as broken. What I found online sounded like a proper mess. Still, if there is something that is broken it seems better that we at least get broken warnings in the test set(?) I have fixing this as relatively high priority.